### PR TITLE
Added telegram service, products pages and services, modal pages

### PR DESCRIPTION
**Pull request by Maximus2806**

### CODE REVIEW

File: src/data/types/toastMessage.types.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export enum TOAST_MESSAGE {
```
Comment (danikkrapivnitskiy) [inline src/data/types/toastMessage.types.ts:1]: product toast message then

File: src/ui/pages/baseModal.page.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { BasePage } from './base.page.js';
+
+export abstract class BaseModalPage extends BasePage {
+  protected readonly 'Submit button' = '//div[@class="modal-footer"]//button[1]';
+  protected readonly 'Cancel button' = '//div[@class="modal-footer"]//button[2]';
+  protected readonly 'Cross button' = '//div[@class="modal-header"]/button';
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/baseModal.page.ts:6]: better verify by text if possible, not by position

File: src/ui/tests/login/auth.setup.ts
Comment (danikkrapivnitskiy) [inline src/ui/tests/login/auth.setup.ts:1]: why removed?

File: src/ui/pages/modals/detailsModal.page.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { BaseModalPage } from '../baseModal.page.js';
+
+export class ModalWindowPage extends BaseModalPage {
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/modals/detailsModal.page.ts:3]: this should be only for Product I think

File: src/data/types/toastMessage.types.ts (hunk 1 above)
Comment (Maximus2806) [reply src/data/types/toastMessage.types.ts:1]: Done. For other pages we just need to add corresponding enums

File: src/ui/pages/modals/detailsModal.page.ts (hunk 3 above)
Comment (Maximus2806) [reply src/ui/pages/modals/detailsModal.page.ts:3]: Renamed the file and class name

File: src/ui/pages/salesPortal.page.ts [hunk 4]
```diff
@@ -2,6 +2,9 @@ import { BasePage } from './base.page.js';
 
 export abstract class SalesPortalPage extends BasePage {
   protected readonly spinner = this.findElement('.spinner-border');
+  protected readonly toast = '#toast .toast-body';
+  protected readonly 'Add New Product button' = 'button.page-title-header';
+  protected readonly 'Close toast message' = '.d-flex button[title="Close"]';
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/salesPortal.page.ts:7]: can it be without `.d-flex`?

Comment (Maximus2806) [reply src/ui/pages/salesPortal.page.ts:7]: Removed .d-flex I hope this selector will be stablewithout it

File: src/ui/services/products/product.service.ts [hunk 5]
```diff
@@ -2,20 +2,20 @@ import { expect, Page } from '@playwright/test';
...
+import { ProductsModalWindowPage } from '../../pages/modals/productDetailsModal.page.js';
 import { AddNewProductPage } from '../../pages/products/addNewProduct.page.js';
 import { ProductsListPage } from '../../pages/products/products.page.js';
 import _ from 'lodash';
 
 export class ProductsListService {
     private productsPage: ProductsListPage;
     private addNewProductPage: AddNewProductPage;
-    private modalWindowPage: ModalWindowPage;
+    private modalWindowPage: ProductsModalWindowPage;
```
Comment (danikkrapivnitskiy) [inline src/ui/services/products/product.service.ts:13]: rename variable

File: src/ui/tests/products/smoke.spec.ts [hunk 6]
```diff
@@ -4,39 +4,39 @@ import { SignInService } from '../../services/signIn.service';
...
-    let salesPortalService: SalesPortalService;  
-  
-  test.beforeEach(async function ({page}) {  
+test.describe('[UI] [Product] Smoke Login - Create -Verify - Delete', async function () {
+  let signInService: SignInService;
+  let homeService: HomeService;
+  let productsService: ProductsListService;
+  let addProductService: AddProductService;
+  let productsPage: ProductsListPage;
+  let salesPortalService: SalesPortalService;
```
Comment (danikkrapivnitskiy) [inline src/ui/tests/products/smoke.spec.ts:18]: use fixture please

File: src/ui/services/products/product.service.ts (hunk 5 above)
Comment (Maximus2806) [reply src/ui/services/products/product.service.ts:13]: Done

File: src/ui/tests/products/smoke.spec.ts (hunk 6 above)
Comment (Maximus2806) [reply src/ui/tests/products/smoke.spec.ts:18]: Done. Added new file ui/tests/products/smoke-fictures.smoke.ts Check please wheter it what you meant

File: src/ui/pages/baseModal.page.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  }
+
+  async clickCross() {
+    await this.click(this['Close modal button']);
+  }
+
+  async clickSubmit() {
+    await this.click(this['Submit button']);
+  }
+}
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/baseModal.page.ts:19]: should it be removed? We will have separate methods for each page, right?

File: src/ui/pages/salesPortal.page.ts [hunk 8]
```diff
@@ -2,6 +2,9 @@ import { BasePage } from './base.page.js';
 
 export abstract class SalesPortalPage extends BasePage {
   protected readonly spinner = this.findElement('.spinner-border');
+  protected readonly toast = '#toast .toast-body';
+  protected readonly 'Add New Product button' = 'button.page-title-header';
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/salesPortal.page.ts:6]: it should be in product page

File: src/ui/services/products/product.service.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+export class ProductsListService {
+  private productsPage: ProductsListPage;
+  private addNewProductPage: AddNewProductPage;
+  private modalWindowPage: ProductDetailsModalPage;
+  private deleteModalPage: DeleteModalPage;
+  constructor(protected page: Page) {
+    (this.productsPage = new ProductsListPage(page)),
+      (this.addNewProductPage = new AddNewProductPage(page)),
+      (this.modalWindowPage = new ProductDetailsModalPage(page)),
+      (this.deleteModalPage = new DeleteModalPage(page));
```
Comment (danikkrapivnitskiy) [inline src/ui/services/products/product.service.ts:19]: did you try run prettier? Extra brackets

File: src/ui/services/salesPortal.service.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { expect } from 'playwright/test';
+import { SalesPortalPage } from '../pages/salesPortal.page.js';
+
+export class SalesPortalService {
+  constructor(private salesPortalPage: SalesPortalPage) {}
+
+  async verifyNotification<T>(expectedMessage: T) {
```
Comment (danikkrapivnitskiy) [inline src/ui/services/salesPortal.service.ts:7]: in your method `verifyNotification` you call `getToastMessage`. Call it in one style please

File: src/ui/tests/products/smoke-fixtures.spec.ts [hunk 11]
```diff
@@ -0,0 +1,27 @@
+import { test } from '../../../fixtures/services.fixture';
+import { generateNewProduct } from '../../../data/products/generateProduct.js';
+import { PRODUCTS_TOAST_MESSAGE } from '../../../data/types/productsToastMessage.types.js';
+import _ from 'lodash';
+import { SalesPortalService } from '../../services/salesPortal.service.js';
+import { ProductsListPage } from '../../pages/products/products.page.js';
+
+test.describe('[UI] [Product] Smoke Login - Create -Verify - Delete', async function () {
+  let productsPage: ProductsListPage;
```
Comment (danikkrapivnitskiy) [inline src/ui/tests/products/smoke-fixtures.spec.ts:9]: add it to fixture also

File: src/ui/tests/products/smoke-fixtures.spec.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+
+  test.beforeEach(async function ({ signInPageService, page }) {
+    productsPage = new ProductsListPage(page);
+    await signInPageService.openSalesPortal();
+  });
+
+  test('Test', async ({ homePageService, productsPageService, addNewProductPageService }) => {
+    const product = generateNewProduct();
+    const salesPortalService = new SalesPortalService(productsPage);
```
Comment (danikkrapivnitskiy) [inline src/ui/tests/products/smoke-fixtures.spec.ts:18]: you don't need to call it, just add new SalesPortalService to fixture and call salesPortalService in test

File: src/ui/tests/products/smoke.spec.ts [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import _ from 'lodash';
+import { SalesPortalService } from '../../services/salesPortal.service.js';
+import { ProductsListPage } from '../../pages/products/products.page.js';
+
+test.describe('[UI] [Product] Smoke Login - Create -Verify - Delete', async function () {
+  let signInService: SignInService;
+  let homeService: HomeService;
+  let productsService: ProductsListService;
+  let addProductService: AddProductService;
+  let productsPage: ProductsListPage;
```
Comment (danikkrapivnitskiy) [inline src/ui/tests/products/smoke.spec.ts:17]: use fixture

Comment (Maximus2806) [reply src/ui/tests/products/smoke.spec.ts:17]: I createted a new file with the same tests but using fixtures so it can be just deleted

Comment (danikkrapivnitskiy) [reply src/ui/tests/products/smoke.spec.ts:17]: make this test be called simply smoke, don't need to specify it as fixture test

File: src/ui/pages/salesPortal.page.ts (hunk 8 above)
Comment (Maximus2806) [reply src/ui/pages/salesPortal.page.ts:6]: Replaced 'Add New Product button'
Toast will be used on other pages so I leave it as it is

File: src/ui/services/products/product.service.ts (hunk 9 above)
Comment (Maximus2806) [reply src/ui/services/products/product.service.ts:19]: fixed

File: src/ui/tests/products/smoke-fixtures.spec.ts (hunk 11 above)
Comment (Maximus2806) [reply src/ui/tests/products/smoke-fixtures.spec.ts:9]: done

File: src/ui/tests/products/smoke-fixtures.spec.ts (hunk 12 above)
Comment (Maximus2806) [reply src/ui/tests/products/smoke-fixtures.spec.ts:18]: Done

File: src/ui/services/salesPortal.service.ts (hunk 10 above)
Comment (Maximus2806) [reply src/ui/services/salesPortal.service.ts:7]: Renamed with getNotificationMessage

File: src/ui/pages/salesPortal.page.ts [hunk 14]
```diff
@@ -11,6 +13,14 @@ export abstract class SalesPortalPage extends BasePage {
   async waitForSpinnerToHide() {
     await this.waitForElement(this.spinner, 'hidden');
   }
+  async getNotificationMessage() {
+    const toastMessage = await this.getText(this.toast);
+    return toastMessage;
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/salesPortal.page.ts:18]: variable names should be also updated

File: src/ui/pages/salesPortal.page.ts [hunk 15]
```diff
@@ -11,6 +13,14 @@ export abstract class SalesPortalPage extends BasePage {
...
+  }
+
+  async closeToastMessage() {
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/salesPortal.page.ts:21]: also this func

File: src/ui/tests/products/smoke.spec.ts (hunk 13 above)
Comment (danikkrapivnitskiy) [reply src/ui/tests/products/smoke.spec.ts:17]: delete it?

File: src/ui/pages/modals/productDetailsModal.page.ts [hunk 16]
```diff
@@ -0,0 +1,31 @@
...
+  
+  async clickActionButton() {    
+    await this.click(this['Submit button']);
+  }
+
+  async clickCancel() {    
+    await this.click(this['Cancel button'])
+  }
+
+  async clickCross() {
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/modals/productDetailsModal.page.ts:16]: leave in one style, at the end of name should be button

File: src/ui/pages/modals/productDetailsModal.page.ts [hunk 17]
```diff
@@ -0,0 +1,31 @@
+import { BaseModalPage } from '../baseModal.page.js';
+
+export class ProductDetailsModalPage extends BaseModalPage {
+  uniqueElement: string;
```
Comment (danikkrapivnitskiy) [inline src/ui/pages/modals/productDetailsModal.page.ts:4]: why do you need it?

File: src/ui/tests/products/smoke.spec.ts (hunk 13 above)
Comment (Maximus2806) [reply src/ui/tests/products/smoke.spec.ts:17]: Deleted files whithout fixtures

File: src/ui/pages/salesPortal.page.ts (hunk 14 above)
Comment (Maximus2806) [reply src/ui/pages/salesPortal.page.ts:18]: Renamed

File: src/ui/pages/salesPortal.page.ts (hunk 15 above)
Comment (Maximus2806) [reply src/ui/pages/salesPortal.page.ts:21]: Done

File: src/ui/pages/modals/productDetailsModal.page.ts (hunk 16 above)
Comment (Maximus2806) [reply src/ui/pages/modals/productDetailsModal.page.ts:16]: Renamed method

File: src/ui/pages/modals/productDetailsModal.page.ts (hunk 17 above)
Comment (Maximus2806) [reply src/ui/pages/modals/productDetailsModal.page.ts:4]: to implement parent method

**Review by danikkrapivnitskiy — APPROVED** [review]
well done

**Pull request merged**